torch/_export/functionalize_assertions.py

```diff
@@ -0,0 +1,52 @@
+import torch
+import copy
+from typing import Dict
+from torch._ops import OpOverload
+
+aten = torch.ops.aten
+
+_NON_FUNCTION_TO_FUNCTIONAL_ASSERTION_FUNCS: Dict[OpOverload, OpOverload] = {
+    aten.sym_constrain_range.default: aten.functional_sym_constrain_range,
+    aten._assert_async.default: aten._functional_assert_async.default,
+    aten._assert_async.msg: aten._functional_assert_async.msg,
+}
+
+
+def functionalize(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
```

Comment:
Can we rename it to something like "_functionalize_side_effectful_ops"? There is another public API called `functionalize` in pytorch.